protocols/example/shoppingcart/products.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "cloudstate/entity_key.proto";
+
+package com.example.shoppingcart;
+
+option go_package = "tck/shoppingcart";
+
+message UpdateCartQuantityRequest {
+    string product_id = 1 [(.cloudstate.entity_key) = true];
+    string user_id = 2;
+    int32 quantity = 3;
+}
+
+message RemoveProductFromCartRequest {
+    string product_id = 1 [(.cloudstate.entity_key) = true];
+    string user_id = 2;
+}
+
+message GetProductRequest {
+    string product_id = 1 [(.cloudstate.entity_key) = true];
+}
+
+message Product {
+    int32 total_quantities = 1;
+    int32 total_carts = 2;
```

Comment:
should this perhaps be uint32?

Reply:
No, never use `uint32` or `uint64` in APIs. Google [recommends strongly against using them](https://cloud.google.com/apis/design/design_patterns#integer_types). The reason for this, is that they can overflow in languages that don't support unsigned integer types, like Java, where `uint32` gets generated as an `int` and `uint64` gets generated as `long`.

Reply:
interesting, thanks.